### PR TITLE
Code improvement using `describe.one`

### LINKED ISSUE #8: Use `describe.one` instead of `match <regex>`
Using a describe block like this:
```
describe <resource> do
  it { should match /1|ON/ }
end
```
Returns a result that might be hard to read to some users:

Screen Shot 2023-03-09 at 2 25 26 PM

Using `describe.one` and splitting the tests in two would provide a cleaner result. Here are the controls currently using the `match <regex>` approach.

- [x] SV-235134
- [x] SV-235135
- [x] SV-235139
- [x] SV-235148
- [x] SV-235154
- [x] SV-235158
- [x] SV-235186
- [x] SV-235187
- [x] SV-235188
- [x] SV-235189
- [x] SV-235190

### PR DESCRIPTION
**Pull request by karikarshivani**
resolves #8

### CODE REVIEW

File: controls/SV-235158.rb [hunk 1]
```diff
@@ -140,12 +140,22 @@
       it { should_not be_empty }
     end
 
-    sql_session.query(query_firewall_users).results.rows.each do |fw_user|
-      describe "USERHOST #{fw_user['userhost']}" do
+    describe.one do
+        sql_session.query(query_firewall_users).results.rows.each do |fw_user|
```
Comment (karikarshivani) [inline controls/SV-235158.rb:144]: I'd recommend moving `describe.one` inside the `.each` loop

File: controls/SV-235148.rb [hunk 2]
```diff
@@ -69,10 +69,13 @@
   SELECT @@ssl_fips_mode;
   )
 
-  ssl_params = sql_session.query(query_ssl_params).results
-
```
Comment (karikarshivani) [inline controls/SV-235148.rb:73]: It seems like the `ssl_params` variable might have been accidentally removed

File: controls/SV-235187.rb [hunk 3]
```diff
@@ -135,7 +135,11 @@
 
   describe '@@ssl_fips_mode' do
     subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it { should cmp 'ON' }
+  end
+  describe '@@ssl_fips_mode' do
+    subject { ssl_params.column('@@ssl_fips_mode').join }
+    it { should cmp '1' }
   end
```
Comment (karikarshivani) [inline controls/SV-235187.rb:143]: Looks like `describe.one` is missing

File: controls/SV-235148.rb [hunk 4]
```diff
@@ -69,10 +69,13 @@
...
+describe.one do
+  describe '@@ssl_fips_mode' do
+    subject { ssl_params.column('@@ssl_fips_mode').join }
+    it { should cmp 'ON' }
+  end
   describe '@@ssl_fips_mode' do
     subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /ON|STRICT/ }
```
Comment (karikarshivani) [inline controls/SV-235148.rb:76]: `end` for `describe.one do` is missing

File: controls/SV-235148.rb [hunk 5]
```diff
@@ -69,10 +69,13 @@
   SELECT @@ssl_fips_mode;
   )
```
Comment (raziz086) [inline controls/SV-235148.rb:71]: fixed by adding the ssl_params variable

File: controls/SV-235148.rb (hunk 4 above)
Comment (raziz086) [reply controls/SV-235148.rb:76]: The end is already there for the describe.one

Comment (karikarshivani) [reply controls/SV-235148.rb:76]: Discussed this later on a call and talked about using `mock://` to check for syntax errors.

File: controls/SV-235188.rb [hunk 6]
```diff
@@ -61,8 +61,9 @@
...
   ssl_params = sql_session.query(query_ssl_params).results
 
+  ssl_fips_mode = ssl_params.column('@@ssl_fips_mode').join
   describe '@@ssl_fips_mode' do
-    subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it "should be ON or STRICT. Got #{ssl_fips_mode}" do
+      expect(ssl_fips_mode).to be_in(['ON', 'STRICT'])
+    end
   end
```
Comment (karikarshivani) [inline controls/SV-235188.rb:69]: `end` missing

File: controls/SV-235187.rb [hunk 7]
```diff
@@ -133,9 +133,11 @@
 
   ssl_params = sql_session.query(query_ssl_params).results
 
+  ssl_fips_mode = ssl_params.column('@@ssl_fips_mode').join
   describe '@@ssl_fips_mode' do
-    subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it "should be ON or STRICT. Got #{ssl_fips_mode}" do
+      expect(ssl_fips_mode).to be_in(['ON', 'STRICT'])
+    end
```
Comment (karikarshivani) [inline controls/SV-235187.rb:140]: The original values seem to be `1` and `ON`

File: controls/SV-235188.rb [hunk 8]
```diff
@@ -61,8 +61,9 @@
 
   ssl_params = sql_session.query(query_ssl_params).results
 
+  ssl_fips_mode = ssl_params.column('@@ssl_fips_mode').join
   describe '@@ssl_fips_mode' do
-    subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it "should be ON or STRICT. Got #{ssl_fips_mode}" do
+      expect(ssl_fips_mode).to be_in(['ON', 'STRICT'])
+    end
```
Comment (karikarshivani) [inline controls/SV-235188.rb:68]: The original values seem to be `1` and `ON`

File: controls/SV-235189.rb [hunk 9]
```diff
@@ -61,8 +61,9 @@
 
   ssl_params = sql_session.query(query_ssl_params).results
 
+  ssl_fips_mode = ssl_params.column('@@ssl_fips_mode').join
   describe '@@ssl_fips_mode' do
-    subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it "should be ON or STRICT. Got #{ssl_fips_mode}" do
+      expect(ssl_fips_mode).to be_in(['ON', 'STRICT'])
+    end
```
Comment (karikarshivani) [inline controls/SV-235189.rb:68]: The original values seem to be `1` and `ON`

File: controls/SV-235190.rb [hunk 10]
```diff
@@ -67,8 +67,10 @@
 
   ssl_params = sql_session.query(query_ssl_params).results
 
+  ssl_fips_mode = ssl_params.column('@@ssl_fips_mode').join
   describe '@@ssl_fips_mode' do
-    subject { ssl_params.column('@@ssl_fips_mode').join }
-    it { should match /1|ON/ }
+    it "should be ON or STRICT. Got #{ssl_fips_mode}" do
+      expect(ssl_fips_mode).to be_in(['ON', 'STRICT'])
+    end
```
Comment (karikarshivani) [inline controls/SV-235190.rb:74]: The original values seem to be `1` and `ON`

File: controls/SV-235158.rb [hunk 11]
```diff
@@ -140,12 +140,11 @@
       it { should_not be_empty }
     end
 
-    sql_session.query(query_firewall_users).results.rows.each do |fw_user|
-      describe "USERHOST #{fw_user['userhost']}" do
-        subject { fw_user }
-        its(['mode']) { should match /LEARNING|DETECTING|PROTECTING/ }
-      end
```
Comment (karikarshivani) [inline controls/SV-235158.rb:147]: @ejaronne The current targets that we have (both hosted and AWS RDS) don't have the `MYSQL_FIREWALL_USERS` table. Should we consider enabling the MySQL Enterprise Firewall to test this control?

File: inspec.lock [hunk 12]
```diff
@@ -0,0 +1,3 @@
+---
+lockfile_version: 1
+depends: []
```
Comment (karikarshivani) [inline inspec.lock:3]: Needs to be deleted to avoid caching issues

File: controls/SV-235188.rb (hunk 6 above)
Comment (karikarshivani) [reply controls/SV-235188.rb:69]: You can use `mock://` to check for syntax errors

**Comment by karikarshivani** [comment]
Replicated the changes in #21 to avoid conflicts

**Pull request closed without merging**